File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if c == '(' or c == '[' or c == '{':
                stack.append(c)
            else:
                if not stack:
                    return False
                left_paran = stack.pop()
                if c == ')' and left_paran != '(':
                    return False 
                elif c == ']' and left_paran != '[':
                    return False 
                elif c == '}' and left_paran != '{':
                    return False
        return False if stack else True

File: Data Structures & Algorithms/simplify-path/submission-1.java
class Solution {
    public String simplifyPath(String path) {
        Deque<String> stack = new LinkedList<>();
        StringBuilder dir = new StringBuilder();
        path += '/';
        for (char c : path.toCharArray()) {
            if (c != '/') {
                dir.append(c);
            } else {
                String currDir = dir.toString();
                if (currDir.equals("..")) {
                    stack.pollLast();
                } else if (currDir.equals(".")) {
//                    Do nothing
                } else if (!currDir.isBlank()){
                    stack.add(currDir);
                }
                dir.setLength(0);
            }
        }
        return "/" + String.join("/", stack);
    }
}